app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Admin::UsersController < ApplicationController
+  before_action :admin_user
+  before_action :load_user, except: %i(index create new)
+
+  def index
+    @user = User.paginate(page: params[:page],
+      per_page: Settings.app.admin.users.per_page).asc_name
+  end
+
+  def destroy
+    @order_confirm = @user.orders.confirm
+    @order_accept = @user.orders.accept
+    if @order_accept.present?
+      flash[:danger] = t "controllers.admin.user.accept",
+        count: @order_accept.count
+    else
+      delete_user_exists_cancel @order_confirm if @order_confirm.present?
+      if @user.destroy
+        flash[:info] = t "controllers.admin.user.success",
+          count: @order_confirm.size
+      else
```

Comment:
destroy fail nhưng vẫn xử lý delete_user_exists_cancel hả e? 
nên đưa vào xử lý cùng lúc trong transaction, đảm bảo thành công thì thành công hết còn fail thì phải rollback lại toàn bộ